Common/Gfx/Util/D2DBitmapLoader.cpp

```diff
@@ -121,6 +121,76 @@ HRESULT D2DBitmapLoader::LoadBitmapFromFile(const Canvas& canvas, D2DBitmap* bit
 	return cleanup(S_OK);
 }
 
+HRESULT D2DBitmapLoader::LoadBitmapFromPluginMeasure(const Canvas& canvas, D2DBitmap* bitmap, PluginImageData& imageData)
+{
+	if (!bitmap) return E_FAIL;
+
+	auto cleanup = [&](HRESULT hr) { return hr; };
+
+	Microsoft::WRL::ComPtr<IWICBitmapSource> source;
+	IWICBitmap* iwicBitmap;
+
+	HRESULT hr = Gfx::Canvas::c_WICFactory->CreateBitmapFromMemory(
+		imageData.width, imageData.height,
+		GUID_WICPixelFormat32bppBGRA,
+		imageData.width * 4 , (imageData.width * imageData.height * 4) * sizeof(unsigned char),
+		reinterpret_cast<BYTE*>(imageData.pixels), &iwicBitmap);
+	if (SUCCEEDED(hr))
+	{
+		hr = ConvertToD2DFormat(iwicBitmap, source);
+	}
+	if (FAILED(hr)) return cleanup(hr);
+

```

Comment:
From here to the end of this function i thought about extracting it to it's own function and call it for both `LoadBitmapFromImage/PluginMeasure`.